core-bundle/src/EventListener/DbafsMetadataSubscriber.php

```diff
@@ -98,4 +105,25 @@ private function supports(AbstractDbafsMetadataEvent $event): bool
     {
         return 'tl_files' === $event->getTable();
     }
+
+    private function getDefaultLocales(): array
+    {
+        if (null !== $this->defaultLocales) {
+            return $this->defaultLocales;
+        }
+
+        $page = $GLOBALS['objPage'] ?? null;
+
+        if (!$page instanceof PageModel) {
+            return [];
+        }
+
+        $locales = [LocaleUtil::formatAsLocale($page->language)];
+
+        if (null !== $page->rootFallbackLanguage) {
+            $locales[] = LocaleUtil::formatAsLocale($page->rootFallbackLanguage);
+        }
+
+        return $this->defaultLocales = array_unique(array_filter($locales));
+    }
```

Comment:
This is pretty much copied from the image studio code. Should we change the globals access to reading from the request attributes instead?

Reply:
I still don't exactly know myself. Guess @aschempp needs to clarify this once and for all and then we need to change it everywhere (if :D).

Reply:
IMHO, we should merge this PR as is. There will be a separate PR dealing with `$GLOBALS['objPage']`.

Reply:
I agree, we should merge as is and then later on deal with all `$GLOBALS['objPage']` usages. Now that https://github.com/contao/contao/pull/3930 is merged it could be changed to 

```diff
-$page = $GLOBALS['objPage'] ?? null;
+$page = $request->attributes->get('pageModel') ?? null;
```

for instance.